From c26dbb7a053ab6b927444e41f34fbbaadc4eb2bd Mon Sep 17 00:00:00 2001
From: Abishek Ganapathy <quic_abishekg@quicinc.com>
Date: Wed, 20 Mar 2024 18:37:34 +0530
Subject: [PATCH] cfg80211: Reset valid_link bitmap after the link is deleted

In 6.6 kernel, cfg80211_remove_link is called only for the links
present as part of wdev->valid_links. The other links as part of MLD
group but not part of connection are not deleted.

Reset wdev->valid_links bitmap after calling the rdev_del_intf_link
API so that the deletion of other links are taken care in the
driver based on the current valid_links count.

Change-Id: I7323d0008fe6347bc2e424887cc7f6fd9213e0cb
Signed-off-by: Abishek Ganapathy <quic_abishekg@quicinc.com>
---
 net/wireless/util.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/net/wireless/util.c b/net/wireless/util.c
index 1783ab9d57a3..a1f018cf4fd6 100644
--- a/net/wireless/util.c
+++ b/net/wireless/util.c
@@ -2659,10 +2659,10 @@ void cfg80211_remove_link(struct wireless_dev *wdev, unsigned int link_id)
 		break;
 	}
 
-	wdev->valid_links &= ~BIT(link_id);
-
 	rdev_del_intf_link(rdev, wdev, link_id);
 
+	wdev->valid_links &= ~BIT(link_id);
+
 	eth_zero_addr(wdev->links[link_id].addr);
 }
 
-- 
2.34.1

